students/ericrosko/session10-final-project/asset_checker.py

```
@@ -0,0 +1,208 @@
+        pass
+
+    def find_project_file(self, start_path, files):
+        '''
+        Finds the first file with the .xcodeproj extension and
+        returns it
+        '''
+        # assert os.path.isdir(start_path) == True
+
+        for name in os.listdir(start_path):
+            fullpath = os.path.join(start_path, name)
+
+            if not isfile(fullpath) and name not in self.ignored_directories:
+                if fullpath.endswith('xcodeproj'):
+                    files.append(fullpath)
+                else:
+                    self.find_project_file(fullpath, files)
+
+    def parse_project_file(self, path, results):
+        assert os.path.isdir(path), "Expected Xcode package file"
+
+        manifest = os.path.join(path, 'project.pbxproj')
+        # print(manifest)
+        assert os.path.isfile(manifest), "Expected manifest text file"
+
+        regexes = self.compiled_regexes()
+
+        doParse = False
+
+        # we only want to get data from a small section of the pbxproj
+        # file.  This is because in a section below it, the terms image.png
+        # and image.jpeg are used except these aren't real files in your
+        # project.  To avoid accidentally including these false positives,
+        # at least for now I'm just parsing the section entitled
+        # /* Begin PBXBuildFile section */
+        # /* End PBXBuildFile section */
+        try:
+            with open(manifest, 'r') as f:
+
+                for line in f:
+                    line = line.rstrip('\n')
+                    if "/* End PBXBuildFile section */" == line:
+                        doParse = False
+                    elif "/* Begin PBXBuildFile section */" == line:
+                        doParse = True
+                    # print(line)
+                    if doParse:
+                        self.parse_line(line, regexes, results)
+                    # print(results)
+
+        except Exception:
```

Good point.  I know better than to swallow exceptions silently.  Here is what I did above:
        except Exception as e:
            raise e

In your code snipped you just do:
except:
print("file read failure")
raise

So I guess you don't need to reference it with 'e'?